apps/web/vibes/soul/primitives/navigation/index.tsx

```diff
@@ -373,26 +431,44 @@ export const Navigation = forwardRef(function Navigation<S extends SearchResult>
                 </Popover.Content>
               </Popover.Portal>
             </Popover.Root>
+          ) : (
+            <Link
+              aria-label={searchLabel}
+              className="rounded-lg p-1.5 ring-primary transition-colors focus-visible:outline-0 focus-visible:ring-2 @4xl:hover:bg-contrast-100"
+              href={searchHref}
+            >
+              <Search size={20} strokeWidth={1} />
+            </Link>
```

Comment:
So what we're doing here is if the action isn't provided we fall back to the href when clicking the search icon? I like it 👍🏻

Reply:
I noticed `searchHref` was required but `searchAction` was not, but if `searchAction` was undefined we hide the icon completely. One solution was just to make `searchHref` optional. I like this solution better.

Reply:
Yeah, that's good. I think later we can actually make `searchAction` be `Action | string` like Next.js 15 Form API does. I think that would be cleanest

Reply:
True!